context.c

```diff
@@ -398,6 +398,52 @@ const struct iio_backend * const iio_backends[] = {
 };
 const unsigned int iio_backends_size = ARRAY_SIZE(iio_backends);
 
+static int iio_context_update_scale_offset(struct iio_context *ctx)
+{
+	const struct iio_attr *attr;
+	struct iio_channel *chn;
+	struct iio_device *dev;
+	unsigned int i, j;
+	int err;
+
+	if (!ctx->ops->read_attr) {
+		/* read_attr callback not implemented - nothing to do. */
+		return 0;
+	}
+
+	for (i = 0; i < ctx->nb_devices; i++) {
+		dev = ctx->devices[i];
+
+		for (j = 0; j < dev->nb_channels; j++) {
+			chn = dev->channels[j];
+
+			attr = iio_channel_find_attr(chn, "scale");
+			if (attr) {
+				err = iio_attr_read_double(attr,
+							   &chn->format.scale);
+				if (err) {
+					chn_perror(chn, err, "Unable to read scale");
+					return err;
+				}
+
+				chn->format.with_scale = true;
+			}
+
+			attr = iio_channel_find_attr(chn, "offset");
+			if (attr) {
+				err = iio_attr_read_double(attr,
+							   &chn->format.offset);
+				if (err) {
+					chn_perror(chn, err, "Unable to read offset");
+					return err;
```

Comment:
didn't checked the code so just sanity checking... There's really no `with_offset` like in the scale case?

Reply:
Well, there's no need, having no offset is the same as having offset == 0.0 (just like having no scale is the same as having scale == 1.0, but it's too late for that).